spyder/api/ipythonconsole.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+This module contains the ipython console API
+"""
+
+from spyder.plugins.ipythonconsole.utils.messagehandler import SpyderMessageHandler
+
+class IPythonAPIMixin:
+    """
+    Mixin class for SpyderPluginWidget to support communication
+    between the widget and the ipython kernels
+    """
+
+    def __init__(self, main=None):
+        """
+        Mixin class for SpyderPluginWidget to support communication
+        between the widget and the ipython kernels
+        """
+        super(IPythonAPIMixin, self).__init__(main)
+        self.main = main
+        self.ipyconsole = main.ipyconsole
+
+    def registerMessageHandler(self, name, func):
```

Comment:
Is this overriding another method named this? If not, it should be named following the standard lowercase_underscore() convention, per our coding style wiki page (linked in the PR template) and PEP 8.